the-collab-lab/tcl-66-smart-shopping-list
# 2. As a user, I want to read all of my shopping lists and the items in them.

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

In our Home component, we imported the SingleList component and then mapped through it so that we would get an individual component per everything in the list. Then as props, we added key, name, and path with the correct path in the data being pulled. We removed the setListPath from being passed into the home component because it is a function and would not properly work when used as a prop.

In the SingleList component, we added in the useStateWithStorage function to properly set the list path and to make the functionality of the handleClick function work. We also added a key prop to the list item.

In the List component we also mapped through the data to create separate ListItem components and added the key and name prop. In the ListItem component we also added a key prop.

The only thing worth noting is that there were some hiccups attempting to render the proper data before realizing there was an error in the schema. Once resolving that, we were golden. 

## Related Issue

closes #2 

## Acceptance Criteria

- [x] The `useShoppingLists` hook is used in `App.jsx` to get the shopping lists a user has access to from the database
- [x] The lists are passed into the `Home` component as a prop named `data`
- [x] The `setListPath` function is passed into the `Home` component so we can update the current list
- [x] The `useShoppingListData` hook is used in `App.jsx` to get the items in the current list from the Firestore database

## Type of Changes

Enhancement

## Updates

### Before
 We forgot to take screenshots before, so we logged out to show what it would look like.

<img width="969" alt="Screenshot 2024-02-07 at 10 56 02 AM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/98231995/c89d2e95-a2fd-4fbb-84c5-9362ccf35e9e">

<img width="970" alt="Screenshot 2024-02-07 at 10 56 36 AM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/98231995/4bdef4f5-301b-4b45-8a95-51219a7e9619">


### After

<img width="972" alt="Screenshot 2024-02-07 at 10 54 03 AM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/98231995/b331ab01-8abc-4164-9352-055cf3a822e1">

<img width="969" alt="Screenshot 2024-02-07 at 10 54 25 AM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/98231995/6749fe6b-d964-419f-a212-337ea5df4552">


## Testing Steps / QA Criteria

1. From the code, type "npm start" in your terminal
2. Click the Sign In button and sign in using the user you have connected to firebase
3. Click the Groceries button
4. Add /list to the url (should look like localhost:3000/list)
5. View the list :)
